sql/core/src/main/scala/org/apache/spark/sql/execution/joins/OuterJoin.scala

```diff
@@ -111,52 +122,43 @@ trait HashOuterJoin {
   // iterator for performance purpose.
 
   protected[this] def leftOuterIterator(
-      key: InternalRow,
       joinedRow: JoinedRow,
       rightIter: Iterable[InternalRow],
       resultProjection: InternalRow => InternalRow): Iterator[InternalRow] = {
     val ret: Iterable[InternalRow] = {
-      if (!key.anyNull) {
-        val temp = if (rightIter != null) {
-          rightIter.collect {
-            case r if boundCondition(joinedRow.withRight(r)) => resultProjection(joinedRow).copy()
```

Comment:
Just pushed an update which removes this use of `flatMap`, `collect` and partial functions for SMJ outer join.  Since SMJ outer join no longer relies on these changes in OuterJoin I'm going to back out all of the other now-unrelated changes to the other outer join operators. We can pursue optimization of those other joins as part of a followup patch.